src/policy/CMSRucioPolicy/algorithms/auto_approve.py

```diff
@@ -92,15 +79,18 @@ def _get_rules_size(rules):
         return False
 
     # Check activity is User AutoApprove
-    if rule_attributes['activity'] != auto_approve_activity:
+    if rule['activity'] != auto_approve_activity:
         return False
 
     # Check if the rule is locked
-    if rule_attributes['locked']:
+    if rule['locked']:
+        return False
+
+    if rule['expires_at'] is None:
         return False
 
-    # Check if the rule lifetime is less than defined threshold
-    if rule_attributes['lifetime'] > rule_lifetime_threshold:
+    lifetime = (rule['expires_at'] - datetime.utcnow()).total_seconds()
```

Comment:
As a result of the above change. Lifetime needs to be calculated.

The rest of the changes are cosmetic. Introduced by the formatter.